compiler/rustc_typeck/src/collect.rs

```diff
@@ -1441,6 +1441,25 @@ fn generics_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::Generics {
                 // of a const parameter type, e.g. `struct Foo<const N: usize, const M: [u8; N]>` is not allowed.
                 None
             } else if tcx.lazy_normalization() {
+                // Only provide backwards declared generics to cg defaults (#86580)
```

Comment:
It would be great to expand this with a code example. Something like

```rust
fn foo<const N, const M: usize = { N + 1 }>
```

and then explain that, when invoked on the `{N + 1}` constant, we extract the prefix `[N]`